TypeError when viewing some event transactions

### Self-Hosted Version

25.7.0

### CPU Architecture

x86_64

### Docker Version

28.3.3

### Docker Compose Version

2.39.1

### Machine Specification

- [x] My system meets the minimum system requirements of Sentry

### Steps to Reproduce

I upgraded my self-hosted instance from 24.9.0 to 25.5.1, and then to 25.7.0.

Now, I'm unable to view some of the transactions from the discover page, both newly generated ones after the upgrade and some older ones. It however works for some transactions in there.

### Expected Result

I'm able to view all transactions successfully.

### Actual Result

Clicking on some transactions takes me to the view traces page ([example](https://sentry.myorg.com/organizations/myorg/discover/trace/62921b5033514e6faadf1e5adcc433c3/?dataset=transactions&eventId=2da5179b60d949758028a6accf33ffa1&field=title&field=project&field=user.display&field=timestamp&id=6&name=&project=6&query=&queryDataset=transaction-like&sort=-timestamp&source=discover&statsPeriod=24h&timestamp=1754308636&topEvents=5&yAxis=count%28%29)). It shows a loader for a few seconds and then shows _"Woof. We failed to load your trace. If you need to yell at someone, send feedback"_.

docker compose logs:
```
web-1    | 11:58:49 [INFO] sentry.access.api: api.access (method='GET' view='sentry.api.endpoints.catchall.CatchallEndpoint' response=404 user_id='15' is_app='False' token_type='None' is_frontend_request='True' organization_id='None' auth_id='None' path='/api/0/subscriptions/myorg/' caller_ip='<REDACTED_IP>' user_agent='Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36 Edg/138.0.0.0' rate_limited='False' rate_limit_category='None' request_duration_seconds=0.01487278938293457 rate_limit_type='DNE' concurrent_limit='None' concurrent_requests='None' reset_time='None' group='None' limit='None' remaining='None')
web-1    | 11:58:49 [WARNING] django.request: Not Found: /api/0/subscriptions/myorg/ (status_code=404 request=<WSGIRequest: GET '/api/0/subscriptions/myorg/'>)
nginx-1  | <REDACTED_IP> - - [04/Aug/2025:11:58:49 +0000] "GET /api/0/subscriptions/myorg/ HTTP/1.1" 404 0 "https://sentry.myorg.com/organizations/myorg/discover/trace/62921b5033514e6faadf1e5adcc433c3/?dataset=transactions&eventId=2da5179b60d949758028a6accf33ffa1&field=title&field=project&field=user.display&field=timestamp&id=6&name=&project=6&query=&queryDataset=transaction-like&sort=-timestamp&source=discover&statsPeriod=24h&timestamp=1754308636&topEvents=5&yAxis=count%28%29" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36 Edg/138.0.0.0" "<REDACTED_IP>"
web-1    | Traceback (most recent call last):
web-1    |   File "/usr/src/sentry/src/sentry/api/base.py", line 317, in handle_exception_with_details
web-1    |     response = self.handle_exception(exc)
web-1    |   File "/.venv/lib/python3.13/site-packages/rest_framework/views.py", line 475, in handle_exception
web-1    |     self.raise_uncaught_exception(exc)
web-1    |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^
web-1    |   File "/.venv/lib/python3.13/site-packages/rest_framework/views.py", line 486, in raise_uncaught_exception
web-1    |     raise exc
web-1    |   File "/usr/src/sentry/src/sentry/api/base.py", line 431, in dispatch
web-1    |     response = handler(request, *args, **kwargs)
web-1    |   File "/usr/src/sentry/src/sentry/api/endpoints/organization_events_trace.py", line 900, in get
web-1    |     self.serialize(
web-1    |     ~~~~~~~~~~~~~~^
web-1    |         limit,
web-1    |         ^^^^^^
web-1    |     ...<6 lines>...
web-1    |         query_source=self.get_request_source(request),
web-1    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
web-1    |     )
web-1    |     ^
web-1    |   File "/usr/src/sentry/src/sentry/api/endpoints/organization_events_trace.py", line 1324, in serialize
web-1    |     self.update_children(subtrace, limit)
web-1    |     ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
web-1    |   File "/usr/src/sentry/src/sentry/api/endpoints/organization_events_trace.py", line 1131, in update_children
web-1    |     parent.children.sort(key=child_sort_key)
web-1    |     ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
web-1    | TypeError: '<' not supported between instances of 'str' and 'float'
web-1    | 11:58:50 [INFO] sentry.access.api: api.access (method='GET' view='sentry.api.endpoints.organization_events_trace.OrganizationEventsTraceEndpoint' response=500 user_id='15' is_app='False' token_type='None' is_frontend_request='True' organization_id='1' auth_id='None' path='/api/0/organizations/myorg/events-trace/62921b5033514e6faadf1e5adcc433c3/' caller_ip='<REDACTED_IP>' user_agent='Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36 Edg/138.0.0.0' rate_limited='False' rate_limit_category='None' request_duration_seconds=0.31406688690185547 rate_limit_type='DNE' concurrent_limit='None' concurrent_requests='None' reset_time='None' group='None' limit='None' remaining='None')
web-1    | 11:58:50 [ERROR] django.request: Internal Server Error: /api/0/organizations/myorg/events-trace/62921b5033514e6faadf1e5adcc433c3/ (status_code=500 request=<WSGIRequest: GET '/api/0/organizations/myorg/events-trace/62921b5033514e6faadf1e5adcc433c3/?limit=10000&targetId=2da5179b60d949758028a6accf33ffa1&timestamp=1754308636'>)
nginx-1  | <REDACTED_IP> - - [04/Aug/2025:11:58:50 +0000] "GET /api/0/organizations/myorg/events-trace/62921b5033514e6faadf1e5adcc433c3/?limit=10000&targetId=2da5179b60d949758028a6accf33ffa1&timestamp=1754308636 HTTP/1.1" 500 72 "https://sentry.myorg.com/organizations/myorg/discover/trace/62921b5033514e6faadf1e5adcc433c3/?dataset=transactions&eventId=2da5179b60d949758028a6accf33ffa1&field=title&field=project&field=user.display&field=timestamp&id=6&name=&project=6&query=&queryDataset=transaction-like&sort=-timestamp&source=discover&statsPeriod=24h&timestamp=1754308636&topEvents=5&yAxis=count%28%29" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36 Edg/138.0.0.0" "<REDACTED_IP>"
```

### Event ID

_No response_

--- a/src/sentry/api/endpoints/organization_events_trace.py
+++ b/src/sentry/api/endpoints/organization_events_trace.py
@@ -483,24 +483,21 @@ def is_root(item: SnubaTransaction) -> bool:
     return item.get("root", "0") == "1"
 
 
-def child_sort_key(item: TraceEvent) -> list[int | str]:
+def child_sort_key(item: TraceEvent) -> tuple[int, int, str, str]:
     if item.fetched_nodestore and item.nodestore_event is not None:
-        return [
+        return (
             item.nodestore_event.data["start_timestamp"],
             item.nodestore_event.data["timestamp"],
-        ]
-    elif item.span_serialized:
-        return [
+            item.event["transaction"],
+            item.event["id"],
+        )
+    else:
+        return (
             item.event["precise.start_ts"],
             item.event["precise.finish_ts"],
             item.event["transaction"],
             item.event["id"],
-        ]
-    else:
-        return [
-            item.event["transaction"],
-            item.event["id"],
-        ]
+        )
 
 
 def count_performance_issues(
@@ -1322,9 +1319,6 @@ class OrganizationEventsTraceEndpoint(OrganizationEventsTraceEndpointBase):
                 orphans.extend(result)
             elif len(roots) > 0:
                 trace_roots = result
-        sentry_sdk.set_context(
-            "trace_root_keys", {"keys": [child_sort_key(trace_root) for trace_root in trace_roots]}
-        )
         # We sort orphans and roots separately because we always want the root(s) as the first element(s)
         trace_roots.sort(key=child_sort_key)
         orphans.sort(key=child_sort_key)

